spring-cloud-kubernetes-examples/kubernetes-leader-election-example/README.md

```diff
@@ -0,0 +1,83 @@
+# Spring Cloud Kubernetes leader election example
+
+## Setting up the Environment
+
+This example uses a Fabric8 Maven Plugin to deploy an application to a Kubernetes cluster.
+To try it locally, download and install `[Minikube](https://kubernetes.io/docs/getting-started-guides/minikube/)`.
+
+Once Minikube is downloaded, start it with the following command:
+```
+minikube start
+```
+
+And configure environment variables to point to the Minikube's Docker daemon:
+```
+eval $(minikube docker-env)
+```
+
+## Overview
+
+Spring Cloud Kubernetes leader election mechanism implements leader election API of Spring Integration using Kubernetes ConfigMap.
+Multiple instances of the same application compete for a leadership of a specified role.
+But only one of them can become a leader and receive an `OnGrantedEvent` application event with a leadership `Context`.
+All the instances periodically try to get a leadership and whichever comes first - becomes a leader.
+The new leader will remain until either its instance disappears from the cluster or it yields its leadership.
+Once the leader is gone, any of the existing instances can become a new leader (even the previous leader if it yielded leadership but stayed in the cluster). 
+And finally, if the leadership is yielded or revoked for some reason, the old leader receives `OnRevokedEvent` application event.
+
+## Example application usage
+
```

Comment:
an overview of the example would be nice

Reply:
There is an overview section above https://github.com/spring-cloud/spring-cloud-kubernetes/pull/199/files#diff-bb328b84480112ea9a1e607a1c1d3ba1R18